File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cur = 0
        cnt = 0
        for i in range(len(nums)):
            if i ==0:
                cur = nums[i]
                cnt = 1
            else:
                if cur == nums[i]:
                    cnt = cnt + 1
                else:
                    cnt = cnt - 1
                    if cnt == 0:
                        cur = nums[i]
                        cnt = 1
        return cur
        

File: Data Structures & Algorithms/climbing-stairs/submission-1.cpp
class Solution {
public:
    vector<int>dp;
    int rec( int cur, int n) {
        if(cur >=n) return cur == n;
        if(dp[cur] != -1) return dp[cur];
        
        return dp[cur] = rec(cur + 1, n) + rec(cur+2, n);
    }
    int climbStairs(int n) { 
        dp.resize(n, -1);
        return rec( 0, n);

    }
};
